llvm/test/Transforms/InstCombine/log-pow.ll

```diff
@@ -12,6 +12,28 @@ define double @log_pow(double %x, double %y) {
   ret double %log
 }
 
+define double @log_powi_const(double %x) {
+; CHECK-LABEL: @log_powi_const(
+; CHECK-NEXT:    [[LOG1:%.*]] = call fast double @llvm.log.f64(double [[X:%.*]])
+; CHECK-NEXT:    [[MUL:%.*]] = fmul fast double [[LOG1]], -3.000000e+00
+; CHECK-NEXT:    ret double [[MUL]]
+;
+  %pow = call fast double @llvm.powi.f64.i32(double %x, i32 -3)
+  %log = call fast double @log(double %pow)
```

Comment:
Requiring fast for this transform seems like overkill, but that's a pre-existing issue that could be relaxed

Reply:
Is there any existing issue about shifting some transformers to non-fast? I could have a look at it.

Reply:
This feels like reassoc?

Reply:
There was a discussion a few years ago about some place where we were checking reassoc when what we really wanted to do was distribution. I don't remember what the outcome was. As I said, I'd really like to have a broadly defined "algebraic-identity" flag, but we're out of bits (as currently implemented).

I think we agree that "-ffast-math" (and -funsafe-math-optimizations) should definitely allow transformations like this, but once you start turning off individual flags (which is important for users who are trying to locally restrict fast-math to solve accuracy problems), it's hard to say which combination should be required from the current set.

With the current "isFast()" check, even disabling contraction blocks the transformation, which is surely not what we want.

Reply:
The current trend is using `reassoc` for these sorts of transformations, but I'm not entirely happy with that logic. I'm putting together a RFC on tweaking the fast math flags and bringing some more clarity to these flags.